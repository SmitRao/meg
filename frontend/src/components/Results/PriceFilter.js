import React from "react";
import { Input, Button, Row, Col } from "antd";
import {convertValuetoCurr} from "./Price";

const InputGroup = Input.Group;

class PriceFilter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {min: 0, max: 0};

    this.handleMinChange = this.handleMinChange.bind(this);
    this.handleMaxChange = this.handleMaxChange.bind(this);
    this.getMinMax = this.getMinMax.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (this.props.clear && prevProps.clear!==this.props.clear) {
      this.setState({min: 0, max: 0})
    }
    if (this.props.currency !== prevProps.currency) {
      this.setState({
        min: convertValuetoCurr(this.state.min, prevProps.currency, this.props.currency),
        max: convertValuetoCurr(this.state.max, prevProps.currency, this.props.currency)
      });
    }
  }

  handleMinChange(event) {
      let value = parseFloat(event.target.value);
      if (Number.isNaN(value)) {
        value = 0;
      }
    // console.log("handleMinChange", this.props.currency, event.target.value, "USD", convertedMinVal)
      this.setState({min: value});
  }

  handleMaxChange(event) {
    let value = parseFloat(event.target.value);
    if (Number.isNaN(value)) {
      value = 0;
    }
    // console.log("handleMaxChange", this.props.currency, event.target.value, "USD", convertedMaxVal)
    this.setState({max: value});
  }

  getMinMax() {
    let values = {
      min: convertValuetoCurr(this.state.min, this.props.currency, "USD"), 
      max: convertValuetoCurr(this.state.max, this.props.currency, "USD")
    }
    this.props.onSet(values);
  }

  render() {
    const minValue = this.state.min === 0 ? "" : this.state.min; 
    const maxValue = this.state.max === 0 ? "" : this.state.max;
    return (
      <InputGroup size="small" className={this.props.className}>
        <Row justify="center" gutter={8}>
          <Col span={7}>
            <Input placeholder="MIN" onChange={this.handleMinChange} id="minInput" maxLength={4} value={minValue}/>
          </Col>
          <Col span={7}>
            <Input placeholder="MAX" onChange={this.handleMaxChange} id="maxInput" maxLength={4} value={maxValue}/>
          </Col>
          <Col span={4}>
            <Button size="small" onClick={this.getMinMax}>SET</Button>
          </Col>
        </Row>
      </InputGroup>
    );
  }
}

export default PriceFilter;
