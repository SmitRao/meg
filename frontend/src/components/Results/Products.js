import React from "react";
import { Typography, Card, Row, Col} from "antd";
import Price from "../Price/Price";

/*Available currencies: verify in ExchangeRatesToCAD.js
  "CAD","AUD","BRL","CNY","EUR","HKD","INR",
  "IDR","JPY","MYR","MXN","NZD","NOK","PEN",
  "RUB","SAR","SGD","ZAR","KRW","SEK","CHF",
  "TWD","THB","TRY","GBP","USD","VND"
*/


const { Meta } = Card;
const { Text } = Typography;

const generateCard = (product) => {
  return (
    <Card
      hoverable
      onClick={()=> window.location.href=product["ProductUrl"]}
      cover={
        <img
          alt={product["ProductName"]}
          src={product["MainImageUrl"]}
        />
      }
    >
      <Meta title={product["BrandName"]} description={product["ProductName"]} />
      {/*TODO: replace currency with currency to display, product_currency from product info*/}
      <Price product_value={product["PriceInEuros"]} product_currency={product["Currency"]} currency={product["Currency"]}/>
      {/* <Text>{product["PriceInEuros"]}</Text> */}
    </Card>
  );
};

const generateGrid = (products) => {
  return (
    <Row gutter={[32,48]}>
      {products.map((product) => {
        return <Col span={6}>{generateCard(product)} </Col>;
      })}
    </Row>
  );
};

class Products extends React.Component {
  render() {
  // console.log(this.props.products);
    return <div>{generateGrid(this.props.products)}</div>;
  }
}

export default Products;
