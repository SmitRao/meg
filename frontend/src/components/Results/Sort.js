import React from "react";
import { Select, Typography } from "antd";

const { Text } = Typography;
const { Option } = Select;

class Sort extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<div>
			<Text strong>Sort By</Text>
        	<Select defaultValue="htl" style={{ width: 160 }} label="Sort By">
        		<Option value="htl">Price High To Low</Option>
        		<Option value="lth">Low to High</Option>
					</Select>
			</div>
			)
	}
}

export default Sort