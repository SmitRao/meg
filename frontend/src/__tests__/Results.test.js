import TestRenderer from 'react-test-renderer';
import PriceFilter from '../components/Results/PriceFilter';
import Products from '../components/Results/Products';
import Sort from '../components/Results/Sort';
import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';


/*
it ('checks the price filter component', () => {
    const price = TestRenderer.create(<PriceFilter />);
    expect(price).toMatchSnapshot();
});

it ('checks the products component', () => {
    const prod = TestRenderer.create(<Products />);
    expect(prod).toMatchSnapshot();
});
*/
it ('checks the sort component', () => {
    const s = TestRenderer.create(<Sort />);
    expect(s).toMatchSnapshot();
});