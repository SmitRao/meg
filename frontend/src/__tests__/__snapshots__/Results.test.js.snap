// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks the sort component 1`] = `
<div>
  <span
    className="ant-typography"
    style={
      Object {
        "WebkitLineClamp": null,
      }
    }
  >
    <strong>
      Sort By
    </strong>
  </span>
  <div
    className="ant-select ant-select-single ant-select-show-arrow"
    label="Sort By"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "width": 160,
      }
    }
  >
    <div
      className="ant-select-selector"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span
        className="ant-select-selection-search"
      >
        <input
          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autoComplete="off"
          className="ant-select-selection-search-input"
          id="rc_select_TEST_OR_SSR"
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          readOnly={true}
          role="combobox"
          style={
            Object {
              "opacity": 0,
            }
          }
          value=""
        />
      </span>
      <span
        className="ant-select-selection-item"
      >
        Price High To Low
      </span>
    </div>
    <span
      aria-hidden={true}
      className="ant-select-arrow"
      onMouseDown={[Function]}
      style={
        Object {
          "WebkitUserSelect": "none",
          "userSelect": "none",
        }
      }
      unselectable="on"
    >
      <span
        aria-label="down"
        className="anticon anticon-down"
        role="img"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="down"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;
